### PR TITLE
Functions: improve existing sample

### PR DESCRIPTION
**Pull request by anguillanneuf**
This addresses user feedback that compares our current Hello World [example](https://cloud.google.com/functions/docs/calling/pubsub#functions_calling_pubsub-python) for Cloud Function with Cloud Pub/Sub with that of [Firebase with Cloud Pub/Sub](https://firebase.google.com/docs/functions/pubsub-events). 

~1. `else` clause actually outputs something _else_.~
2. Show how users can access message attributes, publish time, and message IDs.

### CODE REVIEW

File: functions/helloworld/main.py [hunk 1]
```diff
@@ -80,19 +80,22 @@ def hello_http(request):
...
     """
     import base64
 
-    if 'data' in data:
-        name = base64.b64decode(data['data']).decode('utf-8')
+    if 'data' in event:
+        name = base64.b64decode(event['data']).decode('utf-8')
     else:
-        name = 'World'
+        name = 'World! Something went wrong'
```
Comment (andrewsg) [inline functions/helloworld/main.py]: Is "something went wrong" always true in the case where there's no data, or is it sometimes just a consequence that the user didn't add any? If something really is wrong, can we be more specific for the user's benefit?

Comment (anguillanneuf) [reply functions/helloworld/main.py]: The goal there is to print something other than "World!" Otherwise the output will always be "Hello World!"

Comment (tswast) [reply functions/helloworld/main.py]: If something really did go wrong, maybe raise an exception?

Comment (ace-n) [reply functions/helloworld/main.py]: If that's the case - maybe we should remove this `if-else` logic from this sample? (Though the current logic does allow non-Pub/Sub triggers to invoke the function without issue.)

IMO, we shouldn't be documenting how to handle (relatively rare) GCP-side system failures (on this page at least).

**Comment by anguillanneuf** [comment]
I have decided to just update the sample to show how to access the message ID and publish time. 

"Hello world!" just indicates something is failing. If users see "Hello {something else}!", they can be sure that everything has worked.

**Comment by andrewsg** [comment]
Thanks!

**Pull request merged**